### ISSUE TITLE
[BUG] GEOIP ACL

### ISSUE THREAD

**Issue by barto95100** [tobychui/zoraxy#257]
**Describe the bug**
Problem with ACL and geoip country

**To Reproduce**
country selected is FRANCE and test to access url with this ACL not work (403 Forbidden)
I verify the ip lookup and the ip used is in maxmind FRANCE Country 

If i add manuelly the ip in ACL thé connections work 

![image](https://github.com/user-attachments/assets/afb1b28b-10b5-4c29-9a54-d34944f75193)

What's the source of geoip database use by zoraxy ?

**Comment by tobychui** [comment]
@barto95100 

Source of geoip database is this: 
https://www.jsdelivr.com/package/npm/@ip-location-db/geolite2-asn

There is nothing I can do as this is part of the CI/CD process of automatically update the build in IPv4 and IPv6 mapping of the country code (which changes all the time). I guess if you already know your IP address, you should use IP whitelist instead.

**Comment by barto95100** [comment]
Ok i check the file download from script :
https://github.com/tobychui/zoraxy/blob/main/tools/update_geodb.sh

download file is :  
https://cdn.jsdelivr.net/npm/@ip-location-db/geo-whois-asn-country/geo-whois-asn-country-ipv4.csv

and after lookup most ip range is not present in this file...

why you don't use the maxmind to download the file ? with API 

Maxmind indicate in site :

![CleanShot 2024-07-21 at 20 45 46](https://github.com/user-attachments/assets/ed4e5d87-f407-494c-919b-abc99a931e9e)

Url : https://support.maxmind.com/hc/en-us/articles/4408216129947-Download-and-Update-Databases

I found this for GO and maxmind : https://pkg.go.dev/github.com/oschwald/maxminddb-golang?utm_source=godoc

I don't know if this could help you because I'm not a developer :(

**Comment by tobychui** [comment]
@barto95100 Maxmind dataset license is not compatible with Zoraxy in some of the use cases. 

> The problem is if I test from my cellular mobile the ip change every time

Your ISP should have been assigned a fixed IP address range. You can look up on the internet regarding what possible IPs your ISP is using and use that CIDR as whitelist target. 

I guess we can switch to the db one if the daily one is not too accurate, as I can see your IP address is correctly reported in the other csv file from the list.

![圖片](https://github.com/user-attachments/assets/2c6142ec-fbd1-49fa-bcf7-43bcbecf48f6)

[DB-IP](https://db-ip.com/) Lite Country 	[CC BY 4.0](https://creativecommons.org/licenses/by/4.0/) by [DB-IP](https://db-ip.com/)

**Comment by barto95100** [comment]
YEs possible to move to dbip all my test indicate the dbip is more update to jsdeliver...

or maybe used the maxmind to update 1 time per day

**Comment by tobychui** [comment]
@barto95100 Well as mention earlier, we cannot use Maxmind database due to license issue. But for the user update-able ip table, I will think about it and see how it can be implemented.

**Comment by tobychui** [comment]
I believe this bug was caused by the improper implementation of geoip trie tree and outdated geoIP database. The trie tree bug was fixed a few version back and now Zoraxy support the command `-update_geoip`. I guess this can be closed as alternative solution has been proposed.
